aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp

```diff
@@ -132,5 +132,14 @@ Tensor cos_nested(const Tensor& self) {
   return map_nt(self, at::cos);
 }
 
+Tensor pin_memory_nested(const Tensor& self, c10::optional<Device> device) {
+  if (self.is_pinned(device)) {
+    return self;
+  }
+  auto* nt_input = get_nested_tensor_impl(self);
+  const auto& input_buffer = nt_input->get_buffer();
+  return wrap_buffer(at::_pin_memory(input_buffer, device), nt_input->get_nested_sizes());
```

Comment:
Shouldn't you preserve stride/storage offset here?

Reply:
Added non-contiguous tests to validate this as well.